### PR TITLE
prov/tcp: Bug fixes 

### PR DESCRIPTION
**Pull request by vkrishna**

### CODE REVIEW

**Comment by vkrishna** [comment]
The 4th commit is for addressing the concern in #4133

File: src/windows/osd.c [hunk 1]
```diff
@@ -502,42 +502,40 @@ void freeifaddrs(struct ifaddrs *ifa)
...
 	for (i = 0; i < iov_cnt; i++) {
-		wsa_buf[i].buf = (char *)iovec[i].iov_base;
-		wsa_buf[i].len = iovec[i].iov_len;
+		ret = send(fd, iovec[i].iov_base, iovec[i].iov_len, 0);
+		if (ret >= 0) {
+			size += ret;
+			if (ret == iovec[i].iov_len)
+				continue;
+		}
+		return (ret < 0)? ret: size;
```
Comment (shefty) [inline src/windows/osd.c]: I think this should be:
```
ret = send(...);
if (ret >= 0) {
    size += ret;
    if (ret != iovec[i].iov_len)
        return size;
} else {
    return size ? size : ret;
}

File: src/windows/osd.c [hunk 2]
```diff
@@ -502,42 +502,40 @@ void freeifaddrs(struct ifaddrs *ifa)
...
-		wsa_buf[i].buf = (char *)iovec[i].iov_base;
-		wsa_buf[i].len = iovec[i].iov_len;
+	for (i = 0; i < iov_cnt; i++) {
+		ret = recv(fd, iovec[i].iov_base, iovec[i].iov_len, 0);
+		if (ret >= 0) {
+			size += ret;
+			if (ret == iovec[i].iov_len)
+				continue;
+		}
+		return (ret < 0)? ret: size;
```
Comment (shefty) [inline src/windows/osd.c]: Similar restructure as send path.

File: src/windows/osd.c (hunk 1 above)
Comment (vkrishna) [reply src/windows/osd.c]: Yes this looks right

**Comment by vkrishna** [comment]
Made those changes. Please take a look. Thanks

**Pull request merged**